src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -148,22 +162,32 @@ ConvSolution InitInvokerFactoryNHWC(const ExecutionContext&,
 
     ConvSolution result;
     result.invoker_factory = [ck_args     = CKArgsType{problem},
+                              problem_    = problem,
                               sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](
                                  const std::vector<Kernel>&) mutable {
-        return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
-                   const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+        return [ck_args     = std::move(ck_args),
+                problem_    = std::move(problem_),
```

Comment:
do not capture a large struct like `problem` by value. Rather let's pick the pieces we need. You can compute them outside the lambda and then capture them by value.